internal/middleware/adapter.go

```diff
@@ -0,0 +1,14 @@
+package middleware
+
+import (
+	"net/http"
+)
+
+type Adapter func(http.Handler) http.HandlerFunc
+
+func Adapt(endpoint http.HandlerFunc, middlewares ...Adapter) http.HandlerFunc {
```

Comment:
As this is `chain of responsibility` pattern, I think it should be named `Chain` or something similar

Reply:
Fixed